pkg/kubelet/util/pod_startup_latency_tracker.go

```diff
@@ -112,6 +114,13 @@ func (p *basicPodStartupLatencyTracker) ObservedPodOnWatch(pod *v1.Pod, when tim
 		metrics.PodStartSLIDuration.WithLabelValues().Observe(podStartSLOduration)
 		metrics.PodStartTotalDuration.WithLabelValues().Observe(podStartingDuration.Seconds())
 		state.metricRecorded = true
+		// if is the first Pod with network track the start values
+		// these metrics will help to identify problems with the CNI plugin
+		if !pod.Spec.HostNetwork && !p.firstNetworkPodSeen {
```

Comment:
For my own education, why are we tracking pods with HostNetwork settings? Is it because these pods can only start after CNI initialization is finished, so we are using them as an indirect indicator of CNI init latency? If so why don't we measure the CNI latency directly?

Reply:
> why are we tracking pods with HostNetwork settings? 

just the contrary, only the first Pod without HostNetwork set, that gives us the indirect CNI latency

> If so why don't we measure the CNI latency directly?

This is complex and better explained on this doc

https://docs.google.com/document/d/1Ab2XvcS3koXB50RMhsfxdlXNkej5kW6RyxN1F48uqTs/edit

the TL;DR is that we created a chicken an egg problem and we need to deal with the consequences, existing network readiness is not accurate and the different CNI implementations usually are deployed as daemonset, so the CNI is installed as a hostNetwork Pod, and once it is running all the network pods can start running. ... there are some solutions going on to improve this modifying the CNI spec, but those things can take years in be adopted by all the components involved